model/src/main/java/com/ge/predix/acs/model/Obligation.java

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright 2017 General Electric Company
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package com.ge.predix.acs.model;
+
+import java.util.Collections;
+import java.util.List;
+
+//import javax.validation.constraints.Max;
+//import javax.validation.constraints.Min;
+//import javax.validation.constraints.NotNull;
+
+import io.swagger.annotations.ApiModel;
+
+/**
+ * @author Sebastian Torres Brown
+ * 
+ *         Keti obligations model defined by XACML 3.O
+ *
+ */
+@ApiModel(
+        description = "Obligation augments the interface contract between the policy engine (PDP) and "
+                + "enforcement point (PEP).  In addition to the actual access decision of either 'Permit' or 'Deny', "
+                + "the policy engine may return some supplementary information to the enforcement point in the form"
+                + " of an Obligation. Obligations are a set of operations that must be performed by the PEP "
+                + "in conjunction with an authorization decision, e.g., 'Permit, and in addition fulfill the "
+                + "indicated Obligation(s)' or, alternatively: 'Deny, and in addition fulfill the indicated"
+                + " Obligation(s)'.  If the PEP does not understand, or cannot fulfill, any of the obligations, "
+                + "then it MUST act as if the PDP had returned a 'Deny' authorization decision value.")
+public class Obligation {
+
+    private String id;
+
+    private ObligationType type;
+
+    private Object actionTemplate;
+
+    private List<ActionArgument> actionArguments = Collections.emptyList();
+
+    /**
+     * @return the id
+     */
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * @param id
+     *            the id to set
+     */
+    public void setId(final String id) {
+        this.id = id;
+    }
+
+    /**
+     * @return the type
+     */
+    public ObligationType getType() {
+        return this.type;
+    }
+
+    /**
+     * @param type
+     *            the type to set
+     */
+    public void setType(final ObligationType type) {
+        this.type = type;
+    }
+
+    /**
+     * @return the actionTemplate
+     */
+    public Object getActionTemplate() {
```

Comment:
* is there a reason for this not to be String? At minimum, we need to document that this must be serialize-able to a string. 
* @team is the name `actionTemplate` from XACML - i dont see it. Otherwise, it may be better to chose something else to avoid any confusion with Policy.Target.Action.  how about `constraint`?

Reply:
@sebastiantb proposed just 'template'.  I am ok with it since it's inside the 'obligations' object. anyone have a better suggestion?

Reply:
@sanjeevchopra, @sebastiantb, initially, actionTemplate inside obligationExpression was resolved into action inside obligation.  If we change actionTemplate to just template what are we going to call resolved template?

Reply:
didn't realize that. I'm ok with the current name then, i guess it is scoped by 'obligation'